### PR TITLE
validate: Validate Raw Data, store in DB

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by odscjames** [comment]
https://github.com/openactive/conformance-services/issues/14 for notes - raw data, filter only some results, and it's a % we show on the results page

Technically we don't need validation_passed - you could do validation_results==[] to get the same but setting up this should make for some very easy SQL queries to get the data we need for generating the stat, and some space efficient indexes later.

File: src/lib/validate-raw-data.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+async function validate_raw_data(raw_data) {
+
+    const result = await validator.validate(raw_data.data);
+    const result_filtered = result.filter(r => r.severity === "failure");
+
+    const client = await database_pool.connect();
+    try {
+        await client.query(
+            'UPDATE raw_data SET validation_done= \'t\', validation_results=$1, validation_passed=$2 WHERE id=$3',
+            [JSON.stringify(result_filtered), (result_filtered.length == 0),  raw_data.id]
```
Comment (michaelwood) [inline src/lib/validate-raw-data.js:41]: Your `===` above reminded me we probably want an `===` here too. It is good practice in JS to keep the equivalency(?) consistent .

File: README.md [hunk 2]
```diff
@@ -48,6 +48,15 @@ To run this:
...
 
+### Validate Raw Data
+
+This will validate the raw data using the standard https://github.com/openactive/data-model-validator library
+and save the results back in the database in the`raw_data` table. 
+
+To run this:
+
+`$ node ./src/bin/validate-raw-data.js`
+
```
Comment (michaelwood) [inline README.md:59]: We could really use npm run scripts, the advantage of those is that you avoid encoding the filename/path in lots of places. Though perhaps this is the only place anyway.

**Review by michaelwood — APPROVED** [review]

File: src/lib/validate-raw-data.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/*
+ * TODO This will run only one check at a time - add something so it runs several at once!
+ * But I want to discuss with others best way to do that first
+ */
+async function validate_raw_data_all() {
+
+    const client = await database_pool.connect();
+    try {
+        while(true) {
```
Comment (michaelwood) [inline src/lib/validate-raw-data.js:12]: Just wondering if you're expecting this to be OK to fail?

Comment (odscjames) [reply src/lib/validate-raw-data.js:12]: Not confident, but this applies to many other places and we need a general review of handling error conditions https://github.com/openactive/conformance-services/issues/26

File: README.md (hunk 2 above)
Comment (odscjames) [reply README.md:59]: https://github.com/openactive/conformance-services/issues/27

Comment (odscjames) [reply README.md:59]: When we set up Heroku some more hard coded paths will appear so this is a good point

**Pull request merged**